zio-http/src/main/scala/zhttp/http/Middleware.scala

```diff
@@ -106,7 +106,10 @@ object Middleware {
       case Codec(decoder, encoder)       => http.contramapZIO(decoder(_)).mapZIO(encoder(_))
       case Identity                      => http.asInstanceOf[Http[R, E, AOut, BOut]]
       case Constant(http)                => http
-      case OrElse(self, other)           => self.execute(http).orElse(other.execute(http))
+      case OrElse(self, other)           =>
+        Http.fromOptionFunction { a =>
+          self.execute(http)(a) orElse other.execute(http)(a)
+        }
```

Comment:
This should mean the same imo. Why is the behaviour different?

Reply:
Due to option type in Error we want it to go to next if Its error but don't want to execute other if error is some. fromOptionFunction handles this behaviour.